### PR TITLE
Improvements for Mono & Docker users

### PR DESCRIPTION
**Pull request by mandrean**
* Make Docker shell script executable, fix "autorest: permission denied" issue
* Optimize Docker image, remove unnecessary dependencies and make image a tiny bit smaller
* Update documentation for Mono and Docker usage

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@mandrean__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: Tools/dockerfiles/Dockerfile [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-# all in one RUN to produce less, smaller layers
+ARG AUTOREST_VERSION=0.16.0
+
+ADD autorest /opt/autorest/autorest
+
+# all-in-one RUN to produce smaller image with less layers
 RUN apt-get -qqy update && \
 	apt-get -qqy upgrade && \
-	apt-get -qqy install wget unzip && \
+	apt-get -qqy install bsdtar && \
```
Comment (mandrean) [inline Tools/dockerfiles/Dockerfile]: No need for wget when we have curl, and bsdtar is much smaller + supports piping the data from curl.

**Comment by azurecla** [comment]
__@mandrean__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, AZPRBOT;

**Comment by fearthecowboy** [comment]
@azuresdkci  retest this please

**Comment by fearthecowboy** [comment]
@mandrean this looks fine to me (although, I haven't used the docker stuff, so I may have to trust you on that 😄  )

Once the checks are done, I'll merge it.

Thanks!

**Comment by fearthecowboy** [comment]
Oh, @mandrean ,while I'm thinking of it...

I'm going to move the build to dotnet-cli/CoreCLR tools very soon, before I do, can I send you a link to the branch to help make sure it runs in Docker?

**Comment by mandrean** [comment]
@fearthecowboy Of course! You can e-mail me at sebastian.mandrean@gmail.com

**Pull request merged**